cloudweatherreport/cloudresource/resource.py

```diff
@@ -0,0 +1,153 @@
+import logging
+import os
+import yaml
+
+import cloudweatherreport.cloudresource.aws as aws
+import cloudweatherreport.cloudresource.gce as gce
+import cloudweatherreport.cloudresource.azure as azure
+from cloudweatherreport.utils import juju_cmd
+
+
+class UnknownCloudName(Exception):
+    pass
+
+
+class UnknownCredentialName(Exception):
+    pass
+
+
+class CloudNotSupported(Exception):
+    pass
+
+
+def get_credential_name(creds, cloud_name):
+    cred_name = creds.get('credentials', {}).get(cloud_name, {}).get(
+        'default-credential')
+    if cred_name:
+        return cred_name
+    # If no default credential name, get the first name from the list
+    names = list(creds.get('credentials', {}).get(cloud_name, {}).keys())
```

Comment:
My AWS has a `default-region` key that would conflict with this.  I wonder if `default-credential` and `default-region` are the only reserved keys, or if there's a list somewhere?

Also, if there are more than one credential, this could return a different one each time.  Juju's behavior when there are multiple credentials but no default is to error out; I think we should probably follow that.